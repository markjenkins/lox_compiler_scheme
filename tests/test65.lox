// from craftinginterpreters/test/while/syntax.lox
// modified to use local scope for var c instead of global

// Single-expression body.
{
    var c = 0;
    while (c < 3) print c = c + 1;
}
// expect: 1
// expect: 2
// expect: 3
