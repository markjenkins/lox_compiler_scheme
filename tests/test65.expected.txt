OP_CONSTANT 0
OP_GET_LOCAL 0
OP_CONSTANT 3
OP_LESS
OP_JUMP_IF_FALSE 12
OP_POP
OP_GET_LOCAL 0
OP_CONSTANT 1
OP_ADD
OP_SET_LOCAL 0
OP_PRINT
OP_LOOP 20
OP_POP
OP_POP
OP_RETURN
